x-pack/platform/plugins/shared/osquery/server/plugin.ts

```diff
@@ -151,10 +159,15 @@ export class OsqueryPlugin implements Plugin<OsqueryPluginSetup, OsqueryPluginSt
         if (registerIngestCallback) {
           registerIngestCallback(
             'packagePolicyCreate',
-            async (newPackagePolicy: NewPackagePolicy): Promise<UpdatePackagePolicy> => {
+            async (
+              newPackagePolicy: NewPackagePolicy,
+              _soClient: SavedObjectsClientContract,
```

Comment:
I think you may be able to use that client as a scoped to space client, using request may be invalid, if the package policy is created outside of a request, by a task or fleet setup

Reply:
Good point. Would you know how else can I get current space id? I was only familiar with extracting it from request?

Reply:
You should be able to retrieve it from the saved object client `soClient.getCurrentNamespace`